programs/ido-pool/src/lib.rs

```diff
@@ -104,12 +131,20 @@ pub mod ido_pool {
         let cpi_accounts = Transfer {
             from: ctx.accounts.user_usdc.to_account_info(),
             to: ctx.accounts.pool_usdc.to_account_info(),
-            authority: ctx.accounts.user_authority.clone(),
+            authority: ctx.accounts.user_authority.to_account_info(),
         };
         let cpi_program = ctx.accounts.token_program.to_account_info();
         let cpi_ctx = CpiContext::new(cpi_program, cpi_accounts);
         token::transfer(cpi_ctx, amount)?;
 
+        // Update USDC amounts on the pool account
+        ctx.accounts.pool_account.num_usdc_tokens = (ctx.accounts.pool_account.num_usdc_tokens
+            as u128)
+            .checked_add(amount as u128)
+            .unwrap()
+            .try_into()
+            .unwrap();
```

Comment:
is there a need to cast them into u128s if its just addition? I think you can just do a checked_add between the 2 u64s and it'll have the same effect of guarding against overflow?